key_map.h

```diff
@@ -1,362 +1,379 @@
-  MODIFIERKEY_ALT,  // 130
-  MODIFIERKEY_GUI,  // 131
-  MODIFIERKEY_LEFT_CTRL,  // 132
-  MODIFIERKEY_LEFT_SHIFT,  // 133
-  MODIFIERKEY_LEFT_ALT,  // 134
-  MODIFIERKEY_LEFT_GUI,  // 135
-  MODIFIERKEY_RIGHT_CTRL,  // 136
-  MODIFIERKEY_RIGHT_SHIFT,  // 137
-  MODIFIERKEY_RIGHT_ALT,  // 138
-  MODIFIERKEY_RIGHT_GUI,  // 139
-  KEY_UP,         // 140
-  KEY_DOWN,       // 141
-  KEY_LEFT,       // 142
-  KEY_RIGHT,      // 143
-  KEY_HOME,       // 144
-  KEY_END,        // 145
-  KEY_PAGE_UP,    // 146
-  KEY_PAGE_DOWN,  // 147
-  KEY_INSERT,     // 148
-  KEY_DELETE,     // 149
-  KEY_CAPS_LOCK,  // 150
-  KEY_F1,   // 151
-  KEY_F2,   // 152
-  KEY_F3,   // 153
-  KEY_F4,   // 154
-  KEY_F5,   // 155
-  KEY_F6,   // 156
-  KEY_F7,   // 157
-  KEY_F8,   // 158
-  KEY_F9,   // 159
-  KEY_F10,  // 160
-  KEY_F11,  // 161
-  KEY_F12,  // 162
-  KEY_F13,  // 163
-  KEY_F14,  // 164
-  KEY_F15,  // 165
-  KEY_MENU,  // 166
-  KEY_PRINTSCREEN,  // 167
-  0,  // 168
-  0,  // 169
-  0,  // 170
-  mlc,  // 171
-  mmc,  // 172
-  mrc,  // 173
-  mdc,  // 174
-  msu,  // 175
-  msd,  // 176
-  0,  // 177
-  0,  // 178
-  0,  // 179
-  0,  // 180
-  KEY_SYSTEM_POWER_DOWN,  // 181
-  KEY_SYSTEM_SLEEP,  // 182
-  KEY_SYSTEM_WAKE_UP,  // 183
-  KEY_MEDIA_PLAY,  // 184
-  KEY_MEDIA_PAUSE,  // 185
-  KEY_MEDIA_RECORD,  // 186
-  KEY_MEDIA_FAST_FORWARD,  // 187
-  KEY_MEDIA_REWIND,  // 188
-  KEY_MEDIA_NEXT_TRACK,  // 189
-  KEY_MEDIA_PREV_TRACK,  // 190
-  KEY_MEDIA_STOP,  // 191
-  KEY_MEDIA_EJECT,  // 192
-  KEY_MEDIA_RANDOM_PLAY,  // 193
-  KEY_MEDIA_PLAY_PAUSE,  // 194
-  KEY_MEDIA_PLAY_SKIP,  // 195
-  KEY_MEDIA_MUTE,  // 196
-  KEY_MEDIA_VOLUME_INC,  // 197
-  KEY_MEDIA_VOLUME_DEC,  // 198
-  0,  // 199
-  0,  // 200
-  0,  // 201
-  0,  // 202
-  0,  // 203
-  0,  // 204
-  0,  // 205
-  0,  // 206
-  0,  // 207
-  0,  // 208
-  0,  // 209
-  0,  // 210
-  0,  // 211
-  0,  // 212
-  0,  // 213
-  0,  // 214
-  0,  // 215
-  0,  // 216
-  0,  // 217
-  0,  // 218
-  0,  // 219
-  0,  // 220
-  0,  // 221
-  0,  // 222
-  0,  // 223
-  0,  // 224
-  0,  // 225
-  0,  // 226
-  0,  // 227
-  0,  // 228
-  0,  // 229
-  0,  // 230
-  0,  // 231
-  0,  // 232
-  0,  // 233
-  0,  // 234
-  0,  // 235
-  0,  // 236
-  0,  // 237
-  0,  // 238
-  0,  // 239
-  l0,  // 240
-  l1,  // 241
-  l2,  // 242
-  l3,  // 243
-  l4,  // 244
-  l5,  // 245
-  l6,  // 246
-  l7,  // 247
-  l8,  // 248
-  l9,  // 249
-  0,  // 250
-  0,  // 251
-  0,  // 252
-  0,  // 253
-  fnl,  // 254
-  fn   // 255
-};
-
-#endif
+#ifndef KEY_MAP_H
+#define KEY_MAP_H
+
+#ifndef LAYOUT_US_ENGLISH
+#define LAYOUT_US_ENGLISH
+#endif
+#include "keylayouts.h"
+
+// Be aware. The enum below doesn't include keys like '!' and '{' because they
+// are shift-'1' and shift-'[' respectively.
+enum KeyCode : unsigned char {
+  bks = 8,   // backspace
+  tab = 9,   // tab
+  ent = 13,  // enter
+  esc = 27,  // escape
+  spc = 32,  // or just ' '
+  
+  pls = 199, // keypad plus

```

Comment:
Can you please move this new section down so the enum values are sorted.

Also I'm not sure why this change shows up as a full file modification rather than just added lines. It's really hard to see what exact changes are made. Do you mind submitting a pull request with only the diffs?

Reply:
I added keypad keys with 90b92c0 and credited you in the commit. Hopefully you don't mind.

There were some name changes for consistency and two additional keys: KEYPAD_SLASH and KEYPAD_ENTER.